src/cdk/platform/platform.ts

```diff
@@ -84,5 +84,15 @@ export class Platform {
   /** Whether the current browser is Safari. */
   SAFARI: boolean = this.isBrowser && /safari/i.test(navigator.userAgent) && this.WEBKIT;
 
+  /** Whether the device is a Mac device/OS. */
+  MAC: boolean =
+    !this.isBrowser && /(macintosh|macintel|macppc|mac68k|macos)/i.test(navigator.userAgent);
+
+  /** Whether the device is a Windows device/OS. */
+  WINDOWS: boolean = !this.isBrowser && /(win32|win64|windows|wince)/i.test(navigator.userAgent);
```

Comment:
lets not add the `WINDOWS` / `LINUX` ones unless we find an actual need to test them (your code in this PR only depends on the `MAC` one (also I would call it `MACOS` or `MAC_OS` since the name of the os is macOS)